section3/reflection.md

```diff
@@ -1,13 +1,47 @@
+### 3. In your own words, explain what an Array method is.
+An array method is basically a function used to manipulate an array in some way. For example, the pop method takes out the last element of an array. The unshift method adds an element to the beginning of an array. The MDN documentation provides a heap of array methods that can be used to manipulate the array in many different ways.
 
-6. What questions do you still have about Arrays and loops?
+### 4. What is the purpose of a loop? When we would use a loop in conjunction with array vs. without?
+
+A loop is useful because it gives code the power to repeat whatever function is needed in the program. A loop would be used in conjunction with an array if the code needs to apply a loop to specific elements contained in an array.
+
+For example, if I am using a piece of insurance verification software, I might be prompted to enter my policy number to check if my policy is active. Once I click the submit button, the program will compare the policy number I entered to all of the other policy numbers in their system. These policy numbers are held within an array. The program will loop through the array, and then it will tell me that the plan is active if it finds a matching policy number. I imagine the code would look something like this:
+
+```javascript
+for (var i = 0; i < policies.length; i++) {
+  if (policies[i] === input) {
+    console.log("Plan is active");
+  } else {
+    console.log("No active policy found");
+  }  
+}
+```
+
+Basically, a loop allows for us to repeat any function as many times as we'd like. A loop together with an array is harnessing the repetitive power of the loop, and applying it to specific data contained in the array.
+
+
+
+### 5. If you had to teach a 5 year old about loops, how would you explain it?
+
+Loops are really cool! When you write a loop, you tell your computer to do a task as many times as you tell it to. Let's say you wanted your computer to "sing" Baby Shark to your friends 100 times. You can tell your computer, hey computer, I want you to start at the number 0 and start counting how many times you sing baby shark. I want you to sing Baby Shark 100 times. Then stop. The computer will "sing" Baby Shark 100 times, and your friends will laugh. Cool right?
+
+```javascript
+for (var i = 0; i < 100; i++) {
+  console.log ("BABY SHARK DOO DOO DOO DOO DOO DOODOO!!!!!");
+}
+```
+
+### 6. What questions do you still have about Arrays and loops?
+
+I wonder how complicated arrays and loops can get. I know a little bit about objects in code, and I know that those can be contained in an array. But I want to know more about the specific applications of loops and arrays. Also, I want to know if there are other ways to repeat or group things in code, besides loops and arrays. For example, when I think about insurance policy numbers, that seems like a staggering number of things to be in an array. Is it practical to have arrays that are that massive?
```

Comment:
They can get very complicated, yep, and long.

It is common that you will have nested arrays - so Arrays of Arrays or Arrays that hold Object Literals, etc. 

You'll see this come to life pretty shortly in the first few weeks of Mod 1!